ddtrace/contrib/asgi/middleware.py

```diff
@@ -0,0 +1,113 @@
+from ddtrace import tracer as global_tracer
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.ext import SpanTypes, http
+from ddtrace.http import store_request_headers, store_response_headers
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.settings import config
+from ddtrace.vendor.six.moves import urllib
+
+from ...internal.logger import get_logger
+from .constants import ASGI_SPEC_VERSION, ASGI_VERSION
+from .utils import guarantee_single_callable
+
+log = get_logger(__name__)
+
+config._add("asgi", dict(service_name=config._get_service(default="asgi"), distributed_tracing=True,))
+
+
+def _extract_tags_from_scope(scope):
+    tags = {}
+
+    http_method = scope.get("method")
+    if http_method:
+        tags[http.METHOD] = http_method
+
+    query_string = None
+    if config.asgi.trace_query_string:
+        query_string = scope.get("query_string")
+        if isinstance(query_string, bytes):
+            query_string = query_string.decode()
+        tags[http.QUERY_STRING] = query_string
+
+    server = scope.get("server")
+    if server and len(server) == 2:
+        port = server[1]
+        server_host = server[0] + (":" + str(port) if port != 80 else "")
```

Comment:
I still think it'd be worth being explicit about port 80, but just my opinion 🙂

Reply:
I think this is definitely worth coming to some agreement on to standardize the `http.url` tag across our integrations. 

For example, Django does not include 443 and 80 ports in the URL: 

https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/django/utils.py#L82-L83

Reply:
Agh, okay. Yeah let's be consistent then.